### PR TITLE
Form validation

### PR DESCRIPTION
**Pull request by Kaiserabbas**
I have added the form validation features for the contact form. 
If data is validated input area outline is set to red color. In case of an invalid input entry or no entry, the input outline turns red and an error message appears under the input, providing useful instructions to the user.

### CODE REVIEW

File: form-validation.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const textAreaValue = textArea.value;
+
+  // defining function to check input type errors
+  function checkError(i, message) {
+    const subForm = i.parentElement;
+    const small = subForm.querySelector('small');
+    subForm.className = 'sub-form invalid';
+    small.innerText = message;
+  }
+  // defining function if there's no error
```
Comment (Davidosky007) [inline form-validation.js:22]: * Nice work but I noticed a bug when I enter my email address in capital letters no errors came up below is , Kindly fix it

https://github.com/Kaiserabbas/project-portfolio/assets/52497246/1c9c3e73-803d-4176-a0c3-7bcb62012a86

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Kaiserabbas @mulugithub 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Good looking readme 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kaiserabbas** [comment]
Fixed bug in email validation for capital letter entry. 
image

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Kaiserabbas ,  

Your project is complete! 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- error message well visible


<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**